joomla/joomla-cms
# Language plugin: x-default language

### Description

This patch adds a new `link` tag in the `head` section with `hreflang` attribute set to `x-default`.
### How to test
1. Fresh Joomla install with multilanguage.
2. Apply https://github.com/joomla/joomla-cms/pull/8524 patch and then this patch.
3. Go to language plugin and choose:
   - Item Associations: Yes
   - Add Alternate Meta Tags: Yes
   - Add x-default Meta Tag: Yes
   - X-default language: Choose "Default frontend option" or any language you want as x-default
4. Create language associations between menu items and content
5. Go to frontend, any page with a language association (in all languages) and check the code (`head` tag). There should be a `link`tag with `hreflang` attribute with `x-default` value and the `href`should point to the same content in the language that you selected in "x-default language" field.
6. Play with the plugin options and check if the x-default tag in there or not according to the options you selected.
### B/C

None that i know of.
### More info

> For language/country selectors or auto-redirecting homepages, you should add an annotation for the hreflang value "x-default" as well:
> <link rel="alternate" href="http://example.com/" hreflang="x-default" />

Source: https://support.google.com/webmasters/answer/189077?hl=en
